diff --git a/ppu/include/rsx/gcm_sys.h b/ppu/include/rsx/gcm_sys.h
index 18876f2..bc96bbf 100644
--- a/ppu/include/rsx/gcm_sys.h
+++ b/ppu/include/rsx/gcm_sys.h
@@ -1257,7 +1257,7 @@ void gcmSetDefaultCommandBuffer();
 /*! \brief Obtain registers for controlling the command buffer.
 \return Pointer to a \ref gcmControlRegister structure
 */
-gcmControlRegister* gcmGetControlRegister();
+gcmControlRegister* gcmGetControlRegister(void* context_launder);
 
 s32 gcmInitSystemMode(const u64 mode);
 u32 gcmGetTiledPitchSize(const u32 size);
